docs/index.rst

```diff
@@ -10,51 +10,102 @@
-namedtuple with the following attributes:
 
-============== ======== =======================================
-attribute      type     description
-============== ======== =======================================
-``positions``  float32  coordinates for each atom in the frame
-``time``       float    current system time
-``step``       int      frame index
-``box``        float32  3x3 matrix of the system volume
-============== ======== =======================================
+The TNGFileIterator has attributes related to the trajectory metadata, such as
+the number of integrator steps, the number of steps with data, the block_ids 
+available at each step, and the stride at which each block is written.
+
+The TNGFileIterator returns one frame at a time, which is accessed from the
+:attr:`TNGFileIterator.current_integrator_step` attribute or as part of any
+slicing or indexing operation. A NumPy array of the right size must then be
+provided to a getter method for the data to be read into.
+
+An example of how to read positions from a TNG file is shown below.
 
-For example, the coordinate information can be accessed via the `.positions` 
-attribute of the returned object.
 
 .. code-block:: python
 
-  import pytng
+   import pytng
+   import numpy as np
+
+   with pytng.TNGFileIterator("traj.tng", 'r') as tng:
+
+      # make a numpy array to hold the data using helper function
+      # this array will then be updated in-place 
+      
+      positions = tng.make_ndarray_for_block_from_name("TNG_TRAJ_POSITIONS")
+      
+      # the TNG API uses regular strides for data deposition, here we stride
+      # over the whole trajectory for the frames that have position data
+      # len(tng) is the total number of steps in the file.
+
+      for ts in tng[0:len(tng):tng.block_strides["TNG_TRAJ_POSITIONS"]]:
+         # read the integrator timestep, modifying the current_integrator_step
+         # which is returned as ts
+
+         # get the data from the requested block by supplying NumPy array which
+         # is updated in-place or returned 
+         
+         positons = ts.get_positions(positions)
 
-  with pytng.TNGFile('traj.tng', 'r') as f:
-      for ts in f:
-          ts.positions
+           
 
 It is also possible to slice and index the file object to select particular
 frames
 
 .. code-block:: python
 
   import pytng
+  import numpy as np
 
-  with pytng.TNGFile('traj.tng', 'r') as f:
-      first_frame = f[0]
-      last_frame = f[-1]
+  with pytng.TNGFileIterator('traj.tng', 'r') as tng:
+      tng[100].get_positions(positions)
 
-      every_other_frame = [ts for ts in f[::2]]
+
+Error handling
+==============
+
+PyTNG is designed so that Python level exceptions can (and should) be caught by
+the calling code. However, as the trajectory reading itself is done 
+by Cython at the C level with the GIL released, low level trajectory reading
+errors are caught by more general exceptions at the Python level.
```

Comment:
Are there any specific exceptions that the user needs to be aware of?

Are exceptions documented in the classes (sorry, I am short on time reviewing this)